src/Native/Unix/System.IO.Ports.Native/pal_serial.c

```diff
@@ -32,3 +32,13 @@ intptr_t SystemIoPortsNative_SerialPortOpen(const char * name)
 
     return fd;
 }
+
+int SystemIoPortsNative_SerialPortClose(intptr_t fd)
+{
+    // some devices don't unlock handles from exclusive access
+    // preventing reopening after closing the handle
+
+    // ignoring the error - best effort
+    ioctl(fd, TIOCNXCL);
+    return close(fd);
```

Comment:
Does this compile on all systems?  I'd expect you to need to use the ToFileDescriptor helper.

Reply:
seems like it did on all unixes on the CI

Reply:
Hmm. I'm surprised by that. I think we should use it anyway, if for no other reason than consistency. You can see how we do it elsewhere. Thanks.